### PR TITLE
[GL-1786] Moving Dragmap docker to Warp

### PR DESCRIPTION
**Pull request by cheyenne-gold**
ran test here: https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/15807/console
can validate use of dragmap docker here: https://cromwell.gotc-dev.broadinstitute.org/api/workflows/1/6404eb29-ef20-4e60-81c5-f3a6e778ee68/metadata on line 109

GCR: https://console.cloud.google.com/gcr/images/broad-gotc-prod/US/dragmap
QUAY: https://quay.io/repository/broadinstitute/gotc-prod-dragmap?tag=latest&tab=tags

### CODE REVIEW

File: tasks/broad/DragmapAlignment.wdl [hunk 1]
```diff
@@ -31,15 +31,17 @@ task SamToFastqAndDragmapAndMba {
     Boolean hard_clip_reads = false
     Boolean unmap_contaminant_reads = true
 
+    String docker = "us.gcr.io/broad-gotc-prod/dragmap:1.0.0-1.2.1-2.26.4-1.11-1636046599"
+    Int cpu = 16
     Float disk_multiplier = 8
-    Int memory_gb = 40
+    Int memory_mb = 4000
```
Comment (michaelgatzen) [inline tasks/broad/DragmapAlignment.wdl:37]: Dragmap needs 40 GiB of memory, which is 40,960 MiB
```suggestion
    Int memory_mb = 40960
```

**Review by michaelgatzen — DISMISSED** [review]

File: dockers/broad/dragmap/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        rm -rf /var/cache/yum \
+        ; \
+        \
+# Install TINI
+    mkdir temp; \
+    cd temp; \
+    wget https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini -O /sbin/tini; \
+    chmod +x /sbin/tini \
+    ; \
+    \
```
Comment (wjdingman1) [inline dockers/broad/dragmap/Dockerfile:48]:
```suggestion
    cd temp; \
    \
    wget https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini -O /sbin/tini; \
    chmod +x /sbin/tini \
    ; \
```

File: dockers/broad/dragmap/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    wget ${DRAGMAP_URL}; \
+    tar -xf ${DRAGMAP_VERSION}.tar.gz; \
+    rm ${DRAGMAP_VERSION}.tar.gz; \
+    cd DRAGMAP-${DRAGMAP_VERSION}; \
+    make; \
+    mv build/release/dragen-os /usr/local/bin/; \
+    cd ../; \
+    rm -r DRAGMAP-${DRAGMAP_VERSION} \
+    ; \
+    \
```
Comment (wjdingman1) [inline dockers/broad/dragmap/Dockerfile:59]:
```suggestion
    wget ${DRAGMAP_URL}; \
    tar -xf ${DRAGMAP_VERSION}.tar.gz; \
    rm ${DRAGMAP_VERSION}.tar.gz; \
    \
    cd DRAGMAP-${DRAGMAP_VERSION}; \
    make; \
    mv build/release/dragen-os /usr/local/bin/; \
    \
    cd ../; \
    rm -r DRAGMAP-${DRAGMAP_VERSION} \
    ; \
```

File: dockers/broad/dragmap/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    wget ${SAMTOOLS_URL}; \
+    tar -xf samtools-${SAMTOOLS_VERSION}.tar.bz2; \
+    rm samtools-${SAMTOOLS_VERSION}.tar.bz2; \
+    cd samtools-${SAMTOOLS_VERSION}; \
+    make; \
+    make install; \
+    cd ../; \
+    rm -r samtools-${SAMTOOLS_VERSION} \
+    ; \
+    \
```
Comment (wjdingman1) [inline dockers/broad/dragmap/Dockerfile:70]:
```suggestion
    wget ${SAMTOOLS_URL}; \
    tar -xf samtools-${SAMTOOLS_VERSION}.tar.bz2; \
    rm samtools-${SAMTOOLS_VERSION}.tar.bz2; \
    \
    cd samtools-${SAMTOOLS_VERSION}; \
    make; \
    make install; \
    \
    cd ../; \
    rm -r samtools-${SAMTOOLS_VERSION} \
    ; \
```

File: dockers/broad/dragmap/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+# Install PICARD
+    wget ${PICARD_URL}; \
+    mkdir /picard; \
+    mv picard.jar /picard/ \
+    ; \
+    \ 
+    cd ../; \
+    rm -r temp \
+    ;
```
Comment (wjdingman1) [inline dockers/broad/dragmap/Dockerfile:79]:
```suggestion
# Install PICARD
    wget ${PICARD_URL}; \
    mkdir /picard; \
    mv picard.jar /picard/;  \
    \ 
    cd ../; \
    rm -r temp \
    ; \
# Clean up cached files
    yum clean all; \
    rm -rf /var/cache/yum
```

File: dockers/broad/dragmap/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            ncurses-devel \
+            openssl-devel \
+            wget \
+            xz-devel \
+            zlib-devel \
+        ; \
+        yum clean all; \
+        rm -rf /var/cache/yum \
+        ; \
+        \
```
Comment (wjdingman1) [inline dockers/broad/dragmap/Dockerfile:41]:
```suggestion
```

File: dockers/broad/dragmap/Dockerfile (hunk 5 above)
Comment (wjdingman1) [reply dockers/broad/dragmap/Dockerfile:79]: want to clean up the cached files at the end

File: dockers/broad/dragmap/Dockerfile (hunk 3 above)
Comment (wjdingman1) [reply dockers/broad/dragmap/Dockerfile:59]: its good to logically separate each substep within a specific task with a line break i.e. downloading the tar, building the binary, cleaning up. Its a bit pedantic but its the style that we've chosen so its good to be consistent and makes it a little more readable

**Review by wjdingman1 — APPROVED** [review]
overall great! just small formatting changes in the dockerfile

**Comment by cheyenne-gold** [comment]
retest this please

**Review by jessicaway — APPROVED** [review]

File: dockers/broad/dragmap/README.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+#### `docker pull us.gcr.io/broad-gotc-prod/dragmap:1.0.0-1.2.1-2.26.4-1.11-1636392425`
+`
+
+- __What is this image:__ This image is a centos-based custom image for running DRAGMAP, Picard and SAMTOOLS, it uses `centos:centos7` as a base image.
+- __What are Dragmap, Picard and Samtools:__ Dragmap is the Dragen mapper/aligner Open Source Software, [more info](https://github.com/Illumina/DRAGMAP). Picard is a set of command line tools for manipulating high-throughput sequencing (HTS) data and formats, [more info](https://github.com/broadinstitute/picard). Samtools is a suite of programs for interacting with high-throughput sequencing data. See [here](https://github.com/samtools/samtools) more information.
+- __How to see tool version used in image:__ Please see below.
+
+## Versioning
+
+Dragmap uses the following convention for verisoning:
```
Comment (gbggrant) [inline dockers/broad/dragmap/README.md:16]:
```suggestion
Dragmap uses the following convention for versioning:
```

**Review by gbggrant — APPROVED** [review]

File: dockers/broad/dragmap/README.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
+# Dragmap
+
+## Quick reference
+
+Copy and paste to pull this image
+
+#### `docker pull us.gcr.io/broad-gotc-prod/dragmap:1.0.0-1.2.1-2.26.4-1.11-1636392425`
+`
```
Comment (michaelgatzen) [inline dockers/broad/dragmap/README.md:8]:
```suggestion

```

**Pull request merged**